Libs/MRML/Core/vtkMRMLScalarVolumeDisplayNode.cxx

```diff
@@ -169,6 +169,16 @@ void vtkMRMLScalarVolumeDisplayNode
 ::SetInputToImageDataPipeline(vtkAlgorithmOutput *imageDataConnection)
 {
   this->Threshold->SetInputConnection(imageDataConnection);
+
+  if (this->GetScalarRangeFlag() == vtkMRMLDisplayNode::UseDirectMapping)
+    {
+    // Bypass window/level to show table scalar range directly
+    this->MapToColors->SetInputConnection(imageDataConnection);
```

Comment:
Reconnection can be an expensive operation. Check the current input and only set the input again the the input is actually different (see for example in vtkMRMLVolumeRenderingDisplayableManager.cxx).

Reply:
I always assumed that SetInputConnection would check to see if the input is already connected. The function does return at this point if the connection hasn't changed: [vtkAlgorithm.cxx#L1000-L1009](https://github.com/Kitware/VTK/blob/3a01239340d7f02dfe0ff45a1d0d464437e78e75/Common/ExecutionModel/vtkAlgorithm.cxx#L1000-L1009).

Is the check performed by SetInputConnection expensive?

Reply:
Yes, I think those calls to before the method return propagate information request throughout the entire pipeline. As far as I remember, it made a few fps difference in liver ultrasound display. It would be useful if you could check with a profiler if the check is still needed, as things may have changed since I tested this a few years ago.

Reply:
I didn't see any significant difference between the two options under RelWithDebInfo. In any case, I've added a check to make sure that the input is different.